pkg/apis/core/validation/project.go

```diff
@@ -72,9 +73,24 @@ func ValidateProjectSpec(projectSpec *core.ProjectSpec, fldPath *field.Path) fie
 	}
 	ownerFound := false
 
+	members := make(map[string]struct{}, len(projectSpec.Members))
+
 	for i, member := range projectSpec.Members {
 		idxPath := fldPath.Child("members").Index(i)
 
+		apiGroup, kind, namespace, name, err := ProjectMemberProperties(member)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), member.Name, err.Error()))
+			continue
+		}
+		id := ProjectMemberId(apiGroup, kind, namespace, name)
+
+		if _, ok := members[id]; ok {
+			allErrs = append(allErrs, field.Duplicate(idxPath, member))
```

Comment:
Is this validation logic considered? I tried to update/create projects with duplicate members as described in #2849 but the duplicate members are merged and I don't receive any error about duplicated entries?

Reply:
Yes, it is considered, but the gardener-apiserver auto-merges your duplicates before calling the validation. This is to ensure backwards-compatibility. In a future version, we remove this auto-merge code again, and then duplicates won't be accepted anymore.

Reply:
Ah, yep, I got it.